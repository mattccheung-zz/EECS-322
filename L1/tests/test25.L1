(:go
  (:go
    0 0

    ; Compute the data
    (rdi <- 848484848485)
    (rsi <- 5)
    (rdx <- 7)
    (rcx <- 9)
    (r8 <- 11)
    (r9 <- 13)
    ((mem rsp -8) <- :testFunction_ret)
    (call :testFunction 6)
    :testFunction_ret

    (rdi <- rax)
    (call print 1)

    (return)
  )

  (:testFunction
    6 6

    ((mem rsp 0) <- rdi)
    ((mem rsp 8) <- rsi)
    ((mem rsp 16) <- rdx)
    ((mem rsp 24) <- rcx)
    ((mem rsp 32) <- r8)
    ((mem rsp 40) <- r9)

    (rdi <- (mem rsp 0))
    (call print 1)
    
    (rdi <- (mem rsp 8))
    (call print 1)

    (rdi <- (mem rsp 16))
    (call print 1)
    
    (rdi <- (mem rsp 24))
    (call print 1)

    (rdi <- (mem rsp 32))
    (call print 1)

    (rdi <- (mem rsp 40))
    (call print 1)

    (rax <- 848484848485)
    (rdi <- (mem rsp 0))
    (cjump rdi < 13 :repeat :leave)
    :repeat
    (rdi += 200)
    (rsi <- (mem rsp 8))
    (rdx <- (mem rsp 16))
    (rcx <- (mem rsp 24))
    (r8 <- (mem rsp 32))
    (r9 <- (mem rsp 40))
    (tail-call :testFunction 6)

    :leave
    (return)
  )
)
