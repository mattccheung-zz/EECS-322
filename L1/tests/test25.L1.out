424242424242
2
3
4
5
6
424242424242
